core_backend/app/contents/models.py

```diff
@@ -106,25 +112,35 @@ async def update_content_in_db(
 
 
 async def delete_content_from_db(
+    user_id: str,
     content_id: int,
     asession: AsyncSession,
 ) -> None:
     """
     Deletes a content from the database
     """
-    stmt = delete(ContentDB).where(ContentDB.content_id == content_id)
+    stmt = (
+        delete(ContentDB)
+        .where(ContentDB.user_id == user_id)
+        .where(ContentDB.content_id == content_id)
+    )
     await asession.execute(stmt)
     await asession.commit()
 
 
 async def get_content_from_db(
+    user_id: str,
```

Comment:
Technically, user_id is not necessary here because content_ids are unique - but perhaps a good fail-safe anyway?